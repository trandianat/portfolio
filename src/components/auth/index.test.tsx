import { FunctionComponent, PropsWithChildren, ReactNode } from 'react';
import renderer from 'react-test-renderer';
import Auth from 'components/auth';

jest.mock('@aws-amplify/ui-react', () => ({
  withAuthenticator:
    (Auth: FunctionComponent<PropsWithChildren<{}>>) =>
    ({ children }: { children: ReactNode }) =>
      <Auth>{children}</Auth>,
  useAuthenticator: () => ({ signOut: jest.fn() }),
}));

jest.mock('react-router-dom', () => ({
  Link: ({ children, to }: { children: ReactNode; to: string }) => (
    <a href={to}>{children}</a>
  ),
  useLocation: () => ({ pathname: '/path' }),
}));

describe('Auth', () => {
  it('should match the snapshot', () => {
    expect(
      renderer.create(<Auth>Content shown after login</Auth>)
    ).toMatchSnapshot();
  });
});
