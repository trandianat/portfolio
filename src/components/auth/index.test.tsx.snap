// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth should match the snapshot 1`] = `
[
  .emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  margin-bottom: 24px;
}

<div
    className="emotion-0"
  >
    <a
      href="projects"
    >
      â‰ª Back
    </a>
    <button
      onClick={[MockFunction]}
    >
      Sign out
    </button>
  </div>,
  "Content shown after login",
]
`;
